tests/boards/nrf/comp/src/test.c

```diff
@@ -0,0 +1,289 @@
+	zassert_true(rc == 0, "Cannot configure comparator.");
+
+	rc = comparator_set_trigger_callback(test_dev, test_callback, NULL);
+	zassert_true(rc == 0, "Cannot set callback for comparator.");
+
+	rc = comparator_set_trigger(test_dev, COMPARATOR_TRIGGER_BOTH_EDGES);
+	zassert_true(rc == 0, "Cannot set trigger for comparator.");
+	k_msleep(10);
+
+	counter = 0;
+	gpio_pin_set_dt(&test_pin_1, 0);
+	gpio_pin_set_dt(&test_pin_2, 1);
+	k_msleep(10);
+	zassert_true(counter == 1, "Comparator was not triggered for first threshold cross");
+
+	gpio_pin_set_dt(&test_pin_2, 0);
+	gpio_pin_set_dt(&test_pin_1, 1);
+	k_msleep(10);
+
+	zassert_true(counter == 2, "Comparator was not triggered for second threshold cross");
+}
+
+/**
+ * @brief Configure comparator in differential mode
+ * trigger both edges, event should be detected for falling one
+ */
+
+ZTEST(comparator_runtime_configure, test_comp_config_diff_falling)
+{
+	int rc;
+
+	gpio_pin_set_dt(&test_pin_1, 0);
+	gpio_pin_set_dt(&test_pin_2, 1);
+	comparator_trigger_is_pending(test_dev);
+
+	counter = 0;
+	struct comp_nrf_comp_diff_config config = comp_diff_config;
+
+	config.isource = COMP_NRF_COMP_ISOURCE_5UA;
+	rc = comp_nrf_comp_configure_diff(test_dev, &config);
+
+zassert_true(rc == 0, "Cannot configure comparator.");
+
+	rc = comparator_set_trigger_callback(test_dev, test_callback, NULL);
+	zassert_true(rc == 0, "Cannot set callback for comparator.");
+
+	rc = comparator_set_trigger(test_dev, COMPARATOR_TRIGGER_FALLING_EDGE);
+	zassert_true(rc == 0, "Cannot set trigger for comparator.");
+
+	counter = 0;
+	gpio_pin_set_dt(&test_pin_1, 1);
+	gpio_pin_set_dt(&test_pin_2, 0);
+	k_msleep(10);
+
+	zassert_true(counter == 0, "Comparator was triggered for rising threshold cross");
+
+	gpio_pin_set_dt(&test_pin_1, 0);
+	gpio_pin_set_dt(&test_pin_2, 1);
+	k_msleep(10);
+
+	zassert_true(counter == 1, "Comparator was not triggered for falling threshold cross");
+}
+
+/**
+ * @brief Configure comparator in differential mode
+ * trigger both edges, event should be detected for rising one
+ */
+
+ZTEST(comparator_runtime_configure, test_comp_config_diff_rising)
+{
+	int rc;
+
+	gpio_pin_set_dt(&test_pin_1, 1);
+	gpio_pin_set_dt(&test_pin_2, 0);
+	comparator_trigger_is_pending(test_dev);
+
+	counter = 0;
+	struct comp_nrf_comp_diff_config config = comp_diff_config;
+
+	config.isource = COMP_NRF_COMP_ISOURCE_10UA;
+	rc = comp_nrf_comp_configure_diff(test_dev, &config);
+	zassert_true(rc == 0, "Cannot configure comparator.");
+
+	rc = comparator_set_trigger_callback(test_dev, test_callback, NULL);
+	zassert_true(rc == 0, "Cannot set callback for comparator.");
+
+	rc = comparator_set_trigger(test_dev, COMPARATOR_TRIGGER_RISING_EDGE);
+	zassert_true(rc == 0, "Cannot set trigger for comparator.");
+
+	counter = 0;
+	gpio_pin_set_dt(&test_pin_1, 0);
+	gpio_pin_set_dt(&test_pin_2, 1);
+	k_msleep(10);
+
+	zassert_true(counter == 0, "Comparator was triggered for falling threshold cross");
```

Comment:
`zassert_equal(atomic_get(&counter), 0, "Comparator was triggered for falling threshold cross");`